src/it/scala/com/exasol/cloudetl/kinesis/KinesisImportQueryGeneratorIT.scala

```diff
@@ -0,0 +1,93 @@
+package com.exasol.cloudetl.kinesis
+
+import java.nio.ByteBuffer
+
+import org.testcontainers.containers.localstack.LocalStackContainer
+
+class KinesisImportQueryGeneratorIT extends KinesisAbstractIntegrationTest {
+  final val TEST_TABLE_NAME = "kinesis_table"
+
+  override final def beforeAll(): Unit = {
+    createKinesisStream(2)
+    setupExasol()
+    putRecordsIntoStream()
+    createKinesisMetadataScript()
+    createKinesisImportScript("...")
+    val columns =
+      """sensorId DECIMAL(18,0),
+        |currentTemperature DECIMAL(18,0),
+        |status VARCHAR(100),
+        |kinesis_shard_id VARCHAR(2000),
+        |shard_sequence_number VARCHAR(2000)"""
+    statement.execute(
+      s"""CREATE OR REPLACE TABLE $TEST_TABLE_NAME
+         | ($columns)
+         |""".stripMargin
+    )
+    val assembledJarName = findAssembledJarName()
+    val _ = statement.execute(
+      s"""CREATE OR REPLACE JAVA SET SCRIPT KINESIS_PATH (...)
+         |EMITS (...) AS
+         |     %jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
+         |     %scriptclass com.exasol.cloudetl.kinesis.KinesisImportQueryGenerator;
+         |     %jar /buckets/bfsdefault/default/$assembledJarName;
+         |/
+         |""".stripMargin
+    )
+  }
+
+  private[this] def putRecordsIntoStream(): Unit = {
+    putRecordIntoStream("{\"sensorId\": 17,\"currentTemperature\": 147,\"status\": \"WARN\"}")
+    putRecordIntoStream("{\"sensorId\": 20,\"currentTemperature\": 15,\"status\": \"OK\"}")
+  }
+
+  private[this] def putRecordIntoStream(recordData: String): Unit = {
+    val data = ByteBuffer.wrap(recordData.getBytes())
+    val _ = kinesisClient.putRecord(TEST_STREAM_NAME, data, "partitionKey-1")
+  }
+
+  test("returns data from a shard") {
+    executeKinesisMetadataScript()
+    val expected = List(
+      (17, 147, "WARN", "shardId-000000000000", true),
+      (20, 15, "OK", "shardId-000000000000", true)
+    )
+    assertResultSet(expected)
+    putRecordIntoStream("{\"sensorId\": 67,\"currentTemperature\": 154,\"status\": \"FAIL\"}")
+    putRecordIntoStream("{\"sensorId\": 54,\"currentTemperature\": 4,\"status\": \"OK\"}")
+    executeKinesisMetadataScript()
+    val expected2 = List(
+      (17, 147, "WARN", "shardId-000000000000", true),
+      (20, 15, "OK", "shardId-000000000000", true),
+      (67, 154, "FAIL", "shardId-000000000000", true),
+      (54, 4, "OK", "shardId-000000000000", true),
+    )
+    assertResultSet(expected2)
+  }
```

Comment:
It tests that there are no duplicated values in the table. I can't rely on the order of test methods so it's not really a good idea to split it into 2 tests. What if the second method runs first and inserts the additional values into the stream? Even if I create a new stream for each test method I can't split testing of duplication.

Reply:
Maybe you have other ideas about how to tests it? I want to insert a few values, check them and then insert more and check again that they are unique. And the ideal test would also check that if I run the script the third time, it consumes no more lines and the table is with the same values, but I thought that it's too much for one method

Reply:
Yes, you can do that in 3 separate tests.

```scala
val firstExpectedValues = List(..)
val secondExpectedValues = 

test("first") {
...
assertResultSet(firstExpectedValues)
}

test("second") {
...
assertResultSet(firstExpectedValues ++ secondExpectedValue)
}

test("third") {
executeScript without putting records to stream
assertResultSet(firstExpectedValues ++ secondExpectedValues)
}
```

Since these tests are defined in order, they are run in order.

Reply:
We decided to keep the tests as it is currently.